sympy/stats/stochastic_process_types.py

```diff
@@ -759,3 +766,210 @@ def limiting_distribution(self):
         eqs.append(sum(wi) - 1)
         soln = list(linsolve(eqs, wi))[0]
         return ImmutableMatrix([[sol for sol in soln]])
+
+
+class BernoulliProcess(DiscreteTimeStochasticProcess):
+    """
+    The Bernoulli process consists of repeated
+    independent Bernoulli trials with the same parameter `p`.
+    It's assumed that the probability `p` applies to every
+    trial and that the outcomes of each trial
+    are independent of all the rest. Therefore Bernoulli Processs
+    is Discrete State and Discrete Time Stochastic Process.
+
+    Parameters
+    ==========
+
+    sym: Symbol/string_types
+    p: Proability of getting success
+
+    Examples
+    ========
+
+    >>> from sympy.stats import BernoulliProcess, P, E
+    >>> B = BernoulliProcess("B", 0.7)
+
+    >>> B.state_space
+    FiniteSet(0, 1)
+
+    >>> (B.p).round(2)
+    0.70
+
+    >>> (B.probability_r_success(ntrials=5, rsuccess=5,evaluate=True)).round(2)
+    0.17
+
+    >>> (B.probability_of_rth_success(ntrial=5, rsuccess=3)).round(2)
```

Comment:
Can we do this using `P`? Something like the one suggested at wiki as shown in the screenshot below,
![api_bern](https://user-images.githubusercontent.com/36567889/71678113-5e6b4500-2daa-11ea-8a56-b3dfe26a13a6.png)

Reply:
In fact, one more way to do this is to query, `P(Eq(B[1], x[1]) & Eq(B[2], x[2]) & Eq(B[3], x[3]) . . . . & Eq(B[4], x[4]))`. I would like to have this implemented.

Reply:
This can be used we don't consider all the possible permutations.

Reply:
> In fact, one more way to do this is to query, `P(Eq(B[1], x[1]) & Eq(B[2], x[2]) & Eq(B[3], x[3]) . . . . & Eq(B[4], x[4]))`. I would like to have this implemented.

Okay, then I would change the part of `probability_r_success`

Reply:
> This can be used it won't consider all the possible permutations.

Can you give an example to explain this?

Reply:
Suppose we have 3 trials and 2 success.
then `S = X1 + X2 + X3`
we want `S=2`
Therefore by P&C we get `binomial(3,2) * p**2 * (1-p)`

Reply:
It seems that API was a bit unclear. 
By, `P(Eq(B[1], x[1]) & Eq(B[2], x[2]) & Eq(B[3], x[3]) . . . . & Eq(B[4], x[4]))` I meant that, B[1] should attain x[1], B[2] should attain x[2], and so on. In other words, probability of a particular sequence of events. Probably, the computation of the above query needs multiplication of probabilities of  individual events.
In fact, `S = X1 + X2 + X3` gives an idea of generating a new random variable, S. What does your code produces on doing something like, `B[1] + B[2] + B[3]`.

Reply:
> n fact, `S = X1 + X2 + X3` gives an idea of generating a new random variable, S. What does your code produces on doing something like, `B[1] + B[2] + B[3]`.

Before the last commit I have implemented code for this one.


> By, `P(Eq(B[1], x[1]) & Eq(B[2], x[2]) & Eq(B[3], x[3]) . . . . & Eq(B[4], x[4]))` I meant that, B[1] should attain x[1], B[2] should attain x[2], and so on.

And now by latest commit , I have implemented for this one. Yes, you are correct in this case it is correct to say `p**r*p**(n-r)`

Reply:
> By, `P(Eq(B[1], x[1]) & Eq(B[2], x[2]) & Eq(B[3], x[3]) . . . . & Eq(B[4], x[4]))` I meant that, B[1] should attain x[1], B[2] should attain x[2], and so on. In other words, probability of a particular sequence of events

I think this one would be useful as for a particular sequence, this will work good, and not generating a new rv `S` . For this I also added a docstring, so there is no confusion at user end.